### DISCUSSION TITLE
Tokio for games

### DISCUSSION THREAD

**Discussion by terrarier2111** [tokio-rs/tokio#4004] [Q&A] [answered]
Does it make sense to use tokio for a game running on tcp?
Is there even a way to explicitly tell tokio that you want to switch to the main thread? (this is needed in order for winit to work)

**Comment by Darksonn** [comment, marked as answer]
It can make sense to use Tokio for the networking portions of a game, but generally you would run the game logic on a dedicated thread. If you want to use the main thread for winit with Tokio on a background thread, you can use the techniques outlined [here][1].

[1]: https://tokio.rs/tokio/topics/bridging

**Reply by terrarier2111** [reply]
Thank you!
